cmd/intervalSubsetMatrix/intervalSubsetMatrix.go

```diff
@@ -0,0 +1,97 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"github.com/vertgenlab/gonomics/exception"
+	"github.com/vertgenlab/gonomics/fileio"
+	"github.com/vertgenlab/gonomics/interval"
+	"log"
+	"sort"
+)
+
+func intervalSubsetMatrix(unionFile string, fileListFile string, outFile string) {
+	var err error
+	var i int
+	var j interval.Interval
+	var currOverlaps []interval.Interval
+	var currIntervalChan <-chan interval.Interval
+	var unionIntervals = make([]interval.Interval, 0)
+	var currLineString string
+
+	unionRecChan := interval.GoReadToChan(unionFile)
+	files := fileio.Read(fileListFile)
+	out := fileio.EasyCreate(outFile)
+
+	for curr := range unionRecChan {
+		unionIntervals = append(unionIntervals, curr)
+	}
+	unionTree := interval.BuildTree(unionIntervals)
+
+	var mat = make(map[string][]int) //this will be our output feature matrix. mat[row] => col, where rows are genomic regions, cols are files.
+	for _, u := range unionIntervals {
+		mat[interval.CoordsToString(u)] = make([]int, len(files)) //initialize all rows to zeros based on the number of files. Map is indexed by coords of regions.
+	}
+
+	for i = range files {
+		currIntervalChan = interval.GoReadToChan(files[i])
+		for j = range currIntervalChan {
+			currOverlaps = interval.Query(unionTree, j, "any")
+			if len(currOverlaps) > 0 {
+				mat[interval.CoordsToString(currOverlaps[0])][i] = 1
```

Comment:
Do you need to loop through currOverlaps and mark them all as one?  I guess not because unionTree elements are defined to be a merge of all elements in "files."  Maybe there is a more general solution here?